drivers/shared/executor/executor_linux.go

```diff
@@ -339,10 +339,21 @@ func (l *LibcontainerExecutor) Shutdown(signal string, grace time.Duration) erro
 		case <-time.After(grace):
 			// Force kill all container processes after grace period,
 			// hence `true` argument.
-			return l.container.Signal(os.Kill, true)
+			if err := l.container.Signal(os.Kill, true); err != nil {
+				return err
+			}
 		}
 	} else {
-		return l.container.Signal(os.Kill, true)
+		if err := l.container.Signal(os.Kill, true); err != nil {
+			return err
+		}
+	}
+
+	select {
+	case <-l.userProcExited:
+		return nil
+	case <-time.After(time.Second * 15):
+		return fmt.Errorf("process failed to exit after 15 seconds")
```

Comment:
Presumably, this is needed because we want to run the deferred `container.Destroy()` only after task exited, but it's a bit unintuitive.  If destroy idempotent, it might be nice to be explicit and call it here as well.

Reply:
Hm, looks like Destroy can return an error, so we should probably do something with it as well (log or return... not sure).

Looks like we also Destroy even if we error on an invalid Signal. Seems fine, but it's not clear to me that the behavior is intentional.

I wonder if we should always explicitly call Destroy after the grace period instead of manually sending the kill signal? That might make the code flow more naturally:

1. If container is running && grace > 0 && isValid(signal): container.Signal(); sleep(grace)/exited
2. container.Destroy
3. if !exited after 15s: error